### PR TITLE
Refactor and increase test coverage

### PR DESCRIPTION
**Pull request by gogit**
Refactor and increase test coverage

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,12 @@ on [Keep a CHANGELOG](http://keepachangelog.com/). This project adheres to
 
 ## [Unreleased]
 
+### Added
+
+- EmbeddedArtemisInitializer class
```
Comment (purple52) [inline CHANGELOG.md]: The description of the change in the change log shoudl describe what the change is (and, probably, why), not just what classes have been added.

File: src/main/java/uk/gov/justice/artemis/EmbeddedArtemisInitializer.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
+package uk.gov.justice.artemis;
+
+import java.util.Set;
+
+import org.apache.activemq.artemis.core.security.CheckType;
+import org.apache.activemq.artemis.core.security.Role;
+import org.apache.activemq.artemis.jms.server.embedded.EmbeddedJMS;
+import org.apache.activemq.artemis.spi.core.security.ActiveMQSecurityManager;
+
+public class EmbeddedArtemisInitializer {
```
Comment (purple52) [inline src/main/java/uk/gov/justice/artemis/EmbeddedArtemisInitializer.java]: Utility classes should be final

File: src/main/java/uk/gov/justice/artemis/EmbeddedArtemisInitializer.java [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+    private EmbeddedArtemisInitializer() {}
```
Comment (purple52) [inline src/main/java/uk/gov/justice/artemis/EmbeddedArtemisInitializer.java]: Should add a Javadoc to explain why this is private

File: src/main/java/uk/gov/justice/artemis/EmbeddedArtemisInitializer.java [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    
+    public static EmbeddedJMS initialize(final EmbeddedJMS jmsServer){
```
Comment (purple52) [inline src/main/java/uk/gov/justice/artemis/EmbeddedArtemisInitializer.java]: Please add Javadoc

File: src/main/java/uk/gov/justice/artemis/EmbeddedArtemisInitializer.java [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+        jmsServer.setConfigResourcePath("broker.xml");
+        jmsServer.setSecurityManager(getSecurityManager());
+        return jmsServer;
+    }
+    
+    public static void checkLoggers() {
+        // ensure that the loggers are in the classpath and loaded
```
Comment (purple52) [inline src/main/java/uk/gov/justice/artemis/EmbeddedArtemisInitializer.java]: Remove inline comment and add Javadoc to method

File: src/main/java/uk/gov/justice/artemis/EmbeddedArtemisInitializer.java [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+        org.apache.activemq.artemis.core.server.ActiveMQServerLogger.class.getCanonicalName();
+        org.apache.activemq.artemis.core.server.ActiveMQServerLogger_$logger.class
+                        .getCanonicalName();
+    }
+
+    public static ActiveMQSecurityManager getSecurityManager() {
+        // override the security manager to cater for multiple test configurations
+        // users and passwords
```
Comment (purple52) [inline src/main/java/uk/gov/justice/artemis/EmbeddedArtemisInitializer.java]: Remove inline comment and add Javadoc

File: src/main/java/uk/gov/justice/artemis/EmbeddedJMSServer.java [hunk 7]
```diff
@@ -0,0 +1,50 @@
+package uk.gov.justice.artemis;
+
+import org.apache.activemq.artemis.jms.server.embedded.EmbeddedJMS;
+
+public class EmbeddedJMSServer {
```
Comment (purple52) [inline src/main/java/uk/gov/justice/artemis/EmbeddedJMSServer.java]: Please add Javadoc for the class and methods

File: src/main/java/uk/gov/justice/artemis/ServerPermit.java [hunk 8]
```diff
@@ -0,0 +1,36 @@
+package uk.gov.justice.artemis;
+
+import java.util.concurrent.Semaphore;
+
+/**
+ * 
+ * Permit for starting and stopping the Artemis Server
+ *
+ */
+public class ServerPermit implements AutoCloseable {
```
Comment (purple52) [inline src/main/java/uk/gov/justice/artemis/ServerPermit.java]: Please explain why this is needed in the Javadoc, and add Javadoc to methods

File: src/test/java/uk/gov/justice/artemis/EmbeddedArtemisInitializerTest.java [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+        ActiveMQSecurityManager activeMQSecurityManager =
+                        EmbeddedArtemisInitializer.getSecurityManager();
+        assertNotNull(activeMQSecurityManager);
+        assertTrue(activeMQSecurityManager.validateUser("user", "password"));
+        assertTrue(activeMQSecurityManager.validateUserAndRole("user", "password",
+                        Collections.emptySet(), CheckType.SEND));
+    }
+
+    @Test
+    public void shouldTestPrivateConstructor()
```
Comment (purple52) [inline src/test/java/uk/gov/justice/artemis/EmbeddedArtemisInitializerTest.java]: There is a useful library we use elsewhere for this: 

        <dependency>
            <groupId>net.trajano.commons</groupId>
            <artifactId>commons-testing</artifactId>
            <scope>test</scope>
        </dependency>

assertUtilityClassWellDefined(HeaderConstants.class)

File: CHANGELOG.md (hunk 1 above)
Comment (gogit) [inline CHANGELOG.md]: Added details of refactoring and code responsibilities

File: src/main/java/uk/gov/justice/artemis/EmbeddedArtemisInitializer.java (hunk 2 above)
Comment (gogit) [inline src/main/java/uk/gov/justice/artemis/EmbeddedArtemisInitializer.java]: made the change

File: src/main/java/uk/gov/justice/artemis/EmbeddedArtemisInitializer.java (hunk 3 above)
Comment (gogit) [inline src/main/java/uk/gov/justice/artemis/EmbeddedArtemisInitializer.java]: made the change

File: src/main/java/uk/gov/justice/artemis/EmbeddedArtemisInitializer.java (hunk 5 above)
Comment (gogit) [inline src/main/java/uk/gov/justice/artemis/EmbeddedArtemisInitializer.java]: made the change

File: src/main/java/uk/gov/justice/artemis/EmbeddedArtemisInitializer.java (hunk 6 above)
Comment (gogit) [inline src/main/java/uk/gov/justice/artemis/EmbeddedArtemisInitializer.java]: made the change

File: src/main/java/uk/gov/justice/artemis/EmbeddedJMSServer.java (hunk 7 above)
Comment (gogit) [inline src/main/java/uk/gov/justice/artemis/EmbeddedJMSServer.java]: Added JavaDoc to all the classes

File: src/main/java/uk/gov/justice/artemis/ServerPermit.java (hunk 8 above)
Comment (gogit) [inline src/main/java/uk/gov/justice/artemis/ServerPermit.java]: made the change

File: src/test/java/uk/gov/justice/artemis/EmbeddedArtemisInitializerTest.java (hunk 9 above)
Comment (gogit) [inline src/test/java/uk/gov/justice/artemis/EmbeddedArtemisInitializerTest.java]: Added the dependency and made the change to use the assertion to test utility classes

**Pull request merged**